### ISSUE TITLE
Multiple styles for the same Control type in a single theme from GUI

### ISSUE THREAD

**Issue by seppoday** [godotengine/godot-proposals#2677]
### Describe the project you are working on
Learning godot
### Describe the problem or limitation you are having in your project
Using multiple styles  from GUI (not script) without creating alot of themes
### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Add posibility to "add_stylebox_override" from Godot GUI. For example under Theme there could be list to pick style. IF you dont choose then it will load default one. Or there could be list under Custom Styles property.
### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
<!![Current style](https://user-images.githubusercontent.com/62170071/116700900-9b0f7680-a9c7-11eb-9b94-840e9af282ae.jpg)

or Custom Styles property:

![Bez nazwy-5](https://user-images.githubusercontent.com/62170071/116704247-7cab7a00-a9cb-11eb-8e5e-d9d7e419f6dd.jpg)

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Yes, but I feel that when prototyping GUI I want to have easy access to how it will look like in editor itself. Without using script. Beside that control nodes are often used by non programmers and now it is kinda confusing to use in editor. You can add multiple styles, but you can't use them.
### Is there a reason why this should be core and not an add-on in the asset library?
I think it would improve core usability of themes. Maybe even help to prototype UI.

**Comment by pycbouh** [comment]
> Add posibility to "add_stylebox_override" from Godot GUI. For example under Theme there could be list to pick style. IF you dont choose then it will load default one.

You already can do that. If a control has an overridable theme properties (fonts, styleboxes, colors) then right below the `Theme` section are `Custom ...` sections:

![image](https://user-images.githubusercontent.com/11782833/116701913-217c8600-a9d1-11eb-93be-7b5cd6959098.png)

Also this PR of mine can be useful to further adjust styles and swap them on the fly:
https://github.com/godotengine/godot/pull/47544

**Comment by seppoday** [comment]
> > Add posibility to "add_stylebox_override" from Godot GUI. For example under Theme there could be list to pick style. IF you dont choose then it will load default one.
> 
> You already can do that. If a control has overridable theme properties (fonts, styleboxes, colors) then right below the `Theme` section are `Custom ...` sections:
> 
> ![image](https://user-images.githubusercontent.com/11782833/116701913-217c8600-a9d1-11eb-93be-7b5cd6959098.png)
> 
> Also this PR of mine can be useful to further adjust styles for individual controls and even swap them on the fly:
> [godotengine/godot#47544](https://github.com/godotengine/godot/pull/47544)

Yea, but it still looks like I can have only one style at a time for control node and if I want to try few different ones I need to save every single one of them as a .tres file. Where in my proposal you have everything under one theme and can easly (with one click) swap betwen few different styles and easly add new to list. After that if you want you can use them from code. Am I wrong?

**Comment by pycbouh** [comment]
If you want to completely swap the style of every control, you should use multiple themes instead. You don't need to assign your theme to each individual control either. It can be assigned to the topmost control, or even in the project settings. Then you just swap one theme for another.

If that's what your problem is, then your idea is basically "let's have multiple Themes in one Theme", but there is no need for that when you can just swap them.

If you, however, want to be able to adjust styles of individual controls between several presets, then the PR I've linked will solve that for you.

**Comment by pycbouh** [comment]
> If I use "add_stylebox_override" in code. I can add one of the predefined styles from Theme.

That's not right. You can create `StyleBox` instances completely in code in place.

> But when I am in GUI I can only use styles from different .tres file or override it manualy.

That's what overrides are for, though. To override manually.

> I cant choose styles from Theme attached to that node which is just...weird to me.

Because `Theme`s don't store "styles" they store definitions for existing properties of controls. A `Panel` has the `panel` style and that's what you define in the `Theme` resource. If you want to swap, you swap the `Theme` resource itself, or you add a manual override. While you can define custom properties in a `Theme` resource, controls are not aware of them, they only use what they know they can use.

That said, once again, the linked PR will help you address that. With that you will be able to define custom types in the same `Theme` resource and change that on a control instead. You won't be able to pick from a list, you'd have to type the name in manually, but that's only because `Theme`s propagate and may not be know to the control beforehand.
